### PR TITLE
chore: clean-up logLevel/disableLogs

### LINKED ISSUE #1868: Enhancement: clean-up logLevel/disableLogs
### What version of `Wrangler` are you using?

2.1.4

### What operating system are you using?

Mac

### Describe the Bug

We don't need to prop-drill.

- `LOGGER_LEVELS` in logger.ts needs two new options: `noRoutes: -2` and `none: -1`
- doLog's messageLevel type becomes `Exclude<LoggerLevel, "none" | "noRoutes">,`
- formatMessage's level type becomes  `Exclude<LoggerLevel, "none" | "noRoutes">`

When logLevel comes in via CLI, set the global logger's logLevel.

When we disable Miniflare's route logger, read the global logger's logLevel, and use that instead of `disableLogs`

### PR DESCRIPTION
**Pull request by JacobMGEvans**
Reference issue: 
> LOGGER_LEVELS in logger.ts needs two new options: noRoutes: -2 and none: -1
> - `doLog`'s messageLevel type becomes `Exclude<LoggerLevel, "none" | "noRoutes">`,
> - formatMessage's level type becomes `Exclude<LoggerLevel, "none" | "noRoutes">`
> - When `logLevel` comes in via CLI, set the global logger's `logLevel`.
> 
> When we disable Miniflare's route logger, read the global logger's logLevel, and use that instead of `disableLogs`
> 

fixes #1868 #1892

### CODE REVIEW

**Comment by JacobMGEvans** [comment]
While doing this cleanup I realized we should push the `log-level` flag to the top level flags. The `pages dev` should still have a `log-level` at that command level with a default to `warn`

**Comment by JacobMGEvans** [comment]
> While doing this cleanup I realized we should push the `log-level` flag to the top level flags. The `pages dev` should still have a `log-level` at that command level with a default to `warn`

Will convert to a ticket.

File: packages/wrangler/src/pages/dev.tsx [hunk 1]
```diff
@@ -453,6 +450,8 @@ export const Handler = async ({
 		}
 	}
 
+	// No longer passing the logLevel into the Dev Servers
+	logger.loggerLevel = "warn";
```
Comment (rozenmd) [inline packages/wrangler/src/pages/dev.tsx:454]: This... totally breaks everything else I've said in this review 🤦

**Review by rozenmd — COMMENTED** [review]
Urgh software is hard

File: packages/wrangler/src/logger.ts [hunk 2]
```diff
@@ -3,6 +3,8 @@ import { formatMessagesSync } from "esbuild";
 import { getEnvironmentVariableFactory } from "./environment-variables";
 
 const LOGGER_LEVELS = {
+	noRoutes: -2,
```
Comment (rozenmd) [inline packages/wrangler/src/logger.ts:6]:
```suggestion
	noRoutes: -2,
```

File: packages/wrangler/src/logger.ts [hunk 3]
```diff
@@ -35,13 +37,19 @@ class Logger {
 	warn = (...args: unknown[]) => this.doLog("warn", args);
 	error = (...args: unknown[]) => this.doLog("error", args);
 
-	private doLog(messageLevel: LoggerLevel, args: unknown[]) {
+	private doLog(
+		messageLevel: Exclude<LoggerLevel, "none" | "noRoutes">,
```
Comment (rozenmd) [inline packages/wrangler/src/logger.ts:41]:
```suggestion
		messageLevel: Exclude<LoggerLevel, "none">,
```

File: packages/wrangler/src/logger.ts [hunk 4]
```diff
@@ -35,13 +37,19 @@ class Logger {
...
+		args: unknown[]
+	) {
 		if (LOGGER_LEVELS[this.loggerLevel] >= LOGGER_LEVELS[messageLevel]) {
 			console[messageLevel](this.formatMessage(messageLevel, format(...args)));
 		}
 	}
 
-	private formatMessage(level: LoggerLevel, message: string): string {
+	private formatMessage(
+		level: Exclude<LoggerLevel, "none" | "noRoutes">,
```
Comment (rozenmd) [inline packages/wrangler/src/logger.ts:50]:
```suggestion
		level: Exclude<LoggerLevel, "none">,
```

File: packages/wrangler/src/dev.tsx [hunk 5]
```diff
@@ -335,7 +335,6 @@ export async function startDev(args: StartDevOptions) {
 	let rerender: (node: React.ReactNode) => void | undefined;
 	try {
 		if (args.logLevel) {
-			// we don't define a "none" logLevel, so "error" will do for now.
 			logger.loggerLevel = args.logLevel === "none" ? "error" : args.logLevel;
```
[the remark this answers is not in the document]
Comment (JacobMGEvans) [reply packages/wrangler/src/dev.tsx:338]:
```suggestion
			logger.loggerLevel = args.logLevel;
```

File: packages/wrangler/src/logger.ts (hunk 2 above)
Comment (rozenmd) [inline packages/wrangler/src/logger.ts:6]:
```suggestion
```

File: packages/wrangler/src/dev.tsx [hunk 6]
```diff
@@ -464,7 +462,6 @@ export async function startDev(args: StartDevOptions) {
 
 export async function startApiDev(args: StartDevOptions) {
 	if (args.logLevel) {
-		// we don't define a "none" logLevel, so "error" will do for now.
 		logger.loggerLevel = args.logLevel === "none" ? "error" : args.logLevel;
```
[the remark this answers is not in the document]
Comment (rozenmd) [reply packages/wrangler/src/dev.tsx:465]:
```suggestion
		logger.loggerLevel = args.logLevel;
```

**Review by rozenmd — APPROVED** [review]

**Pull request merged**